tests/handlers/tests.py

```diff
@@ -1,12 +1,68 @@
+import sys
+import traceback
+from io import BytesIO
+from wsgiref import simple_server
+
+
 from django.core.exceptions import ImproperlyConfigured
 from django.core.handlers.wsgi import WSGIHandler, WSGIRequest, get_script_name
+from django.core.servers.basehttp import get_internal_wsgi_application
 from django.core.signals import request_finished, request_started
 from django.db import close_old_connections, connection
 from django.test import (
     RequestFactory, SimpleTestCase, TransactionTestCase, override_settings,
 )
 from django.utils.version import PY37
 
+from .views import FILE_RESPONSE_HOLDER
+
+# If data is too large, socket will choke, so write chunks no larger than 32MB
+# at a time. The rationale behind the 32MB can be found in #5596#comment:4.
+MAX_SOCKET_CHUNK_SIZE = 32 * 1024 * 1024  # 32 MB
+
+
+class FileWrapperHandler(simple_server.ServerHandler):
```

Comment:
I think we can move these tests to the `builtin_server.tests.WSGIFileWrapperTests` and reuse `builtin_server.tests.FileWrapperHandler`.

Reply:
We can simplify `FileWrapperHandler` only to necessary parts, e.g.
```python
class DummyHandler:
    def log_request(self, *args, **kwargs):
        pass


class FileWrapperHandler(simple_server.ServerHandler):
    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        self.request_handler = DummyHandler()
        self._used_sendfile = False

    def sendfile(self):
        self._used_sendfile = True
        return True
```
What do you think?

Reply:
Fine with me.